src/main/java/org/folio/circulation/services/PubSubPublishingService.java

```diff
@@ -22,16 +24,15 @@
   private static final Logger logger = LoggerFactory.getLogger(PubSubPublishingService.class);
 
   private final Map<String, String> okapiHeaders;
-  private final PubsubClient pubSubClient;
+  private final Context vertxContext;
 
   public PubSubPublishingService(RoutingContext routingContext) {
     this(new WebContext(routingContext));
   }
 
   public PubSubPublishingService(WebContext context) {
-    this.pubSubClient = new PubsubClient(context.getOkapiLocation(), context.getTenantId(),
-      context.getOkapiToken());
     this.okapiHeaders = context.getHeaders();
+    vertxContext = Vertx.currentContext();
```

Comment:
Maybe we can introduce a new method in the WebContext class:

```
  public Vertx getVertx() {
    return routingContext.vertx();
  }
```

Just to be 100% sure that the vertx is reused.

Reply:
It is better to get the current context than get one from the routing context because they can be different

Reply:
Thanks @TarasSpashchenko. @alexanderkurash, just ignore my comment.